plugins/python/pyhelpers.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * SPDX-License-Identifier: ISC
+ *
+ * Copyright (c) 2019 Robert Manner <robert.manner@oneidentity.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/*
+ * This is an open source non-commercial project. Dear PVS-Studio, please check it.
+ * PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+ */
+
+#include "pyhelpers.h"
+
+#include <pwd.h>
+#include <signal.h>
+
+static int _sudo_printf_noop(int msg_type, const char *fmt, ...)
+{
+    (void) msg_type;
+    (void) fmt;
+    return 0;
+}
+
+struct PythonContext py_ctx = {
+    &_sudo_printf_noop,
+    NULL,
+    0,
+    NULL
+};
+
+
+int
+py_is_sudo_log_available(void)
+{
+    debug_decl(py_is_sudo_log_available, PYTHON_DEBUG_INTERNAL);
+    debug_return_int(py_ctx.sudo_log != &_sudo_printf_noop);
+}
+
+char *
+py_join_str_list(PyObject *py_str_list, const char *separator)
+{
+    debug_decl(py_join_str_list, PYTHON_DEBUG_INTERNAL);
+
+    char *result = NULL;
+    PyObject* p_separator = NULL;
+    PyObject* p_str = NULL;
```

Comment:
What i have meant here is not pointer, but "py_separator", I just mistakenly written "p_". I used py_ prefix for PyObjects because a lot of functions need a variable for the C and the Python thing also which holds the same value.
Should I change these to scheme "py_separator" -> "separator_py" ? Or py_separator is good enough?

Reply:
Sorry, I didn't noticed that my comment got mangled by markdown.  All I meant was to use `PyObject *p_separator` instead of `PyObject* p_separator` so that the star is next to the variable name and not the type.

Reply:
ok, I misunderstood, but at least now I have fixed the py_ prefix as well...